### ISSUE TITLE
Smaller docker images

### ISSUE THREAD

**Issue by pierreozoux** [cozy-labs/cozy-docker#6]
Yep!

Checking this now.

It would be amazing to have smaller images.

I imagine something like:
- a running cozy
- a running cozy with ssl
- a running cozy with ssl and couchdb and emails

I really love the way discourse is handling things:
https://github.com/discourse/discourse_docker/

I'll PR this week if I have time!

Cheers!

**Comment by frankrousseau** [comment]
Thank you @pierreozoux for reporting. I let @Kloadut and @nledez see what we can do for that one. And yes, feel free to contribute :)

**Comment by spiroid** [comment]
@frankrousseau, @pierreozoux: i started some work to have smaller docker images, see : https://forum.cozy.io/t/docker-smaller-images/1162.

Any feedback and contribution are appreciated :) thx.

**Comment by pierreozoux** [comment]
Yes, didn't have time at the end :/ Thanks a lot @spiroid! I'll comment on the forum :)

**Comment by frankrousseau** [comment]
:)

**Comment by Yamakaky** [comment]
You could also use [alpine linux](https://hub.docker.com/_/alpine/) as a base image, it would really slime down the size of the containers.

**Comment by frankrousseau** [comment]
@nledez @kosssi What do you thing about Alpine images?

**Comment by kosssi** [comment]
I love it :heart: 
It's optimization so we can do it after but all my cloud docker is make with Alpine ;)

**Comment by Yamakaky** [comment]
I was thinking: you should not force the user to use the containers. Fox example, on my server, there is already nginx, I don't need another. Maybe then keep the full container and propose slim containers as began by @spiroid with a Compose file. It would be easier to integrate cozy in an existing stack.

**Comment by kosssi** [comment]
yep but

**Comment by kosssi** [comment]
Yes it's the idea but for now it's not so easy to split ;)

**Comment by nledez** [comment]
Hello @Yamakaky,

For now the Docker image exist to simplify testing.

We work on docker-compose, but it's not ready for publish. We focus on a better Debian packaging.

And for alpine subject. This is a wonderfull distribution for simple apps. But Cozy may have some dependencies who come from external apps.

We take the party to support only Debian & Ubuntu for the momment.

And we plan to publish a docker-compose with separated services. We can't anounce a release date. But I can be very happy if community create one before.

**Comment by spiroid** [comment]
hello @Yamakaky, @nledez 

If you're interested in splitting the docker image in smaller images, we already initiated the work with several cozy-\* images.
I also created a repository with a docker-compose file to tie them together : https://github.com/spiroid/cozy

It is working and i use it for my personal cozy instance.
For now it really lacks documentation and is clearly my next task.

You could also have a look at that forum page : https://forum.cozy.io/t/docker-smaller-images/1162 that i'll keep updated when new featues / fixes will be added.

**Comment by spiroid** [comment]
@Yamakaky : you can clearly run this stack without the nginx part by removing it from the docker-compose file.

I'll add a docker-compose file to run the stack without nginx :)
